web-react-ts/src/pages/campaigns/multiplication/MulltiplicationCampaignServiceForm.tsx

```diff
@@ -0,0 +1,353 @@
+  crusadeDate: string
+  preacher: string
+  crusadeLocation: string
+  attendance: string
+  cediIncome: string
+  foreignCurrency: string
+  souls: string
+  miracles: string
+  crusadePictures: string[]
+  treasurers: string[]
+  treasurerSelfie: string
+}
+
+const MultiplicationCampaignServiceForm = ({
+  church,
+  churchId,
+  churchType,
+  RecordServiceMutation,
+}: ServiceFormProps) => {
+  const { clickCard } = useContext(ChurchContext)
+  const navigate = useNavigate()
+
+  const initialValues: FormOptions = {
+    crusadeDate: new Date().toISOString().slice(0, 10),
+    preacher: '',
+    crusadeLocation: '',
+    attendance: '',
+    cediIncome: '',
+    foreignCurrency: '',
+    souls: '',
+    miracles: '',
+    crusadePictures: [''],
+    treasurers: [''],
+    treasurerSelfie: '',
+  }
+
+  const today = new Date()
+
+  const validationSchema = Yup.object({
+    crusadeDate: Yup.date()
+      .max(new Date(), 'Service could not possibly have happened after today')
+      .min(getMondayThisWeek(today), 'You can only fill forms for this week')
+      .required('Date is a required field'),
+    preacher: Yup.string().required('Please pick a name from the dropdown'),
+    crusadeLocation: Yup.string().required(
+      'You cannot submit this form without entering the location of crusade'
+    ),
+    attendance: Yup.number()
+      .typeError('Please enter a valid number')
+      .positive()
+      .integer('You cannot have attendance with decimals!')
+      .required('You cannot submit this form without entering your attendance'),
+    cediIncome: Yup.number()
+      .typeError('Please enter a valid number')
+      .positive()
+      .required('You cannot submit this form without entering your income'),
+    foreignCurrency: Yup.string(),
+    souls: Yup.number()
+      .typeError('Please enter a valid number')
+      .integer('You cannot enter decimals here')
+      .required(
+        'You cannot submit this form without entering your number of souls won'
+      ),
+    miracles: Yup.number()
+      .typeError('Please enter a valid number')
+      .integer('You cannot enter decimals here')
+      .required(
+        'You cannot submit this form without entering your number of miracles'
+      ),
+    crusadePictures: Yup.array()
+      .min(10, 'You must have at least 10 event pictures')
+      .of(Yup.string().required('You must have at least 10 event pictures')),
+    treasurerSelfie: Yup.string().required('You must take a treasurers selfie'),
+    treasurers: Yup.array()
+      .min(2, 'You must have at least two treasurers')
+      .of(Yup.string().required('Please pick a name from the dropdown')),
+  })
+
+  const checkIfArrayHasRepeatingValues = (array: any[]) => {
+    const sortedArray = array.sort()
+    for (let i = 0; i < sortedArray.length - 1; i++) {
+      if (sortedArray[i + 1] === sortedArray[i]) {
+        return true
+      }
+    }
+    return false
+  }
+
+  const parseForeignCurrency = (string: string) => {
```

Comment:
You should have extracted this to an external file so that you import the function so that your code is written in a more reusable way

Reply:
There was no need to redefine the function